pyEpiabm/pyEpiabm/core/place.py

```diff
@@ -105,14 +127,35 @@ def remove_person(self, person: Person):
         if person not in self.persons:
             raise KeyError("Person not found in this place")
         else:
+            group_index = self.get_group_index(person)
+            self.person_groups[group_index].remove(person)
             person.remove_place(self)
             self.persons.remove(person)
 
-    def empty_place(self):
-        """Remove all people from place. For example
+    def get_group_index(self, person):
+        """Get the group of a person in the place.
+
+        :param person: Person associated with group
+        :type person: Person
+        """
+        place_list = [i[0] for i in person.places]
+        ind = place_list.index(self)
+        return person.places[ind][1]
```

Comment:
Why not just search the `person_group` attribute of self? I'm confused why the place is searching though the person's list of places, rather than their own list of people, but I might be missing something here

Reply:
I think I did have it that way originally, but my implementation of searching a dictionary was longer than this. Have put in a check to ensure person is in place though.